docs/src/proposals/fee_transaction_priority.md

```diff
@@ -0,0 +1,210 @@
+    }
+```
+
+3. In the case of a `NoConflict` we run:
+
+```
+    for account_key in transaction_accounts {
+        self.locked_accounts.insert_reference(account_key.key());
+    }
+
+    banking_thread_channel.send(next_highest_transaction);
+```
+
+4. In the case of a `Conflict` we run:
+
+```
+for locked_account_key in transaction_accounts {
+    let account_key = locked_account_key.key()
+    let blocked_transaction_entry = self.blocked_transaction_queues_by_accounts.entry(account_key);
+    match blocked_transaction_entry {
+        Occupied(existing_blocked_transaction) => {
+            // If there is already a set of transactions blocked on this account, add
+            // this transaction to the priority queue.
+            existing_blocked_transaction.insert_transaction(next_highest_transaction);
+        }
+
+        Vacant(vacant_entry) => {
+            // Create a new queue blocked on this transaction
+            let new_blocked_transaction_queue =
+                Rc::new(BlockedTransactionsQueue {
+                    highest_priority_blocked_transaction: next_highest_transaction,
+                    other_blocked_transactions: BinaryHeap::new(),
+                });
+            // Insert into the hashmap for this `account_key`
+            vacant_entry.insert(new_blocked_transaction_queue.clone());
+            // Insert into the `blocked_transactions` hashmap to indicate this set of transactions
+            // is blocked by `next_highest_transaction`
+            self.blocked_transactions.insert(
+                next_highest_transaction.signature(),
+                new_blocked_transaction_queue
+            );
+        }
+    }
+}
+```
+
+5. Run until all `N` BankingStage threads have been sent `processing_batch` transactions (i.e. hit step 3 above).
+
+#### Banking Threads
+1. Banking threads maintain a queue of transactions sent to them by the scheduler, sorted by priority.
+2. Because the scheduler has guaranteed that there are no locking conflicts, the banking thread can process
+some `M` of these transactions at a time and pack them into entries
+
+#### Handling Completion Signals from BankingStage Threads
+
+Outside of the main loop above, we rely on BankingThreads threads signaling us they've finished their
+task to schedule the next transactions.
+
+1. Once a BankingStage thread finishes processing a batch of transactions `completed_transactions_batch` ,
+it sends the `completed_transactions_batch` back to the scheduler via the same channel to signal of completion.
+
+2. Upon receiving this signal, the BankingStage thread processes the locked accounts
+`transaction_accounts` for each `completed_transaction` in `completed_transactions_batch`:
+```
+let mut unlocked_accounts = vec![];
+// First remove all the locks from the tracking list
+for locked_account in transaction_accounts {
+    if self.locked_accounts.remove_reference(locked_account) {
+        unlocked_accounts.push(locked_account.key());
+    }
+}
```

Comment:
fyi, I'm proposing an alternative drastic change, departing from the batching altogether: https://github.com/solana-labs/solana/issues/23548

if the scheduler thread still don't unlock at all until the whole batched (completed) transactions are returned back from the banking stage, i think we still suffer from somewhat constrained tps due to the problem described there (or i might be wrong...)

Reply:
Agreed. What I think will happen is that people participating in a Raydium IDO or NFT drop will be submitting larger numbers of fee-prioritized transactions and these will heavily drown at the ability of other transactions in the pool to be executed in parallel, and will result in the same tps drops. It'll also be tough to deal with these acute spamming periods with the congestion fee raises since it will take time to increase the fees... Maximizing throughput/parallelism as much as possible will more quickly solve demand issues and relieve the spamming more quickly.

Reply:
> Maximizing throughput/parallelism as much as possible will more quickly solve demand issues and relieve the spamming more quickly

gm! https://en.wikipedia.org/wiki/Induced_demand

Reply:
Hmm yeah understandable but I think induced demand > reduced demand due to not meeting expectations that have been set. I was seeing blocks during congestion that had 0 votes or non-votes at times which seems strange (Zan was the first one who spotted one and then I saw others thereafter).

Reply:
> spamming periods with the congestion fee raises since it will take time to increase the fees.

well, I don't think Solana's priority fee is completely same thing as the eth gas. eth gas is for global fee market and solana is for local fee market. and the former is under the persistently saturated condition. the later is under the temporal spiky condition. that's why i want bidding info to be on chain so that people can react quickly to the demand.

> It seems like supply is going down with a lot of demand. I was seeing blocks during congestion that had 0 votes or non-votes at times which seems strange

This is true. but this is a bug, not design by intention. Along with this proposal and #23548 and #21883, we're trying to localize the heavily contended accounts, while payments can be as fast and cheat as possible par solana's selling point. :)

Reply:
Ah yeah, agree with you that it seems like it's a bug-type of situation. With the current situation in which parallelization seems to become compromised though (probably?) it essentially turns what should be a local fee market into a global fee market though I think. That's why I worry about the situation where you have an nft drop -> NFT people prioritize their transactions -> parallelism drops with the batch design -> people outside of the nft drop need to add more priority than the NFT people. It looks like the newer transaction scheduler should help deal with this situation better though so that's good to see.